### PR TITLE
Modernise edk2-platforms (replace EFI_D_* and __FUNCTION__ with DEBUG_* and __func__)

### PR DESCRIPTION
**Pull request by bcran**
Update code to be more C11 compliant by using `__func__`
    
`__FUNCTION__` is a pre-standard extension that gcc and Visual C++ among
others support, while `__func__` was standardized in C99.

Since it's more standard, replace `__FUNCTION__` with `__func__` throughout
edk2-platforms.

------

Replace deprecated EFI_D_* with DEBUG_*
    
Replaced the deprecated EFI_D_{INFO,WARN,ERROR,VERBOSE} usage with
DEBUG_{INFO,WARN,ERROR,VERBOSE}.

### CODE REVIEW

**Review by abdattar — APPROVED** [review]

**Review by samimujawar — APPROVED** [review]
For Platform/ARM/*
Reviewed-by: Sami Mujawar <sami.mujawar@arm.com>

**Review by nhivp — APPROVED** [review]
For Silicon/Ampere and Platform/Ampere, looks good to me. Thanks!

**Review by nate-desimone — APPROVED** [review]
Looks good for Platform/Intel & Silicon/Intel

Thanks @bcran!

**Comment by bcran** [comment]
Waiting on code owner review from @JohnAZoidberg, @ardbiesheuvel, @hrw, @kilaterlee, @leiflindholm, @lixianglai, @ndhillonm, @niruiyu, @vlsunil, @wojtas-marcin, and/or 1 others

**Review by ndhillonm — APPROVED** [review]

**Review by ardbiesheuvel — APPROVED** [review]

**Comment by hrw** [comment]
For */Qemu/SbsaQemu:

Reviewed-by: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>

**Review by hrw — APPROVED** [review]

**Review by leiflindholm — APPROVED** [review]
I'm not sure you need to wait for more approvals for this non-functional change.

**Comment by bcran** [comment]
> I'm not sure you need to wait for more approvals for this non-functional change.

@mdkinney It looks like merging is blocked waiting for additional reviews.

> Waiting on code owner review from JohnAZoidberg, kilaterlee, lixianglai, niruiyu, and/or vlsunil.

> Merging is blocked
> Merging can be performed automatically with 1 approving review.

**Comment by mdkinney** [comment]
A CODEOWNERS file is present, but I do not see any code owner tags in this review.  Is the CODEOWNERS file consistent?  Perhaps it is being ignored due to missing contributors or files not covered.

**Comment by mdkinney** [comment]
There is also something not right about the CODEOWNERS config in this repo.  There should be an icon next to each assign reviewer that is a CODEOWNER.  See this PR as an example: https://github.com/tianocore/edk2-codereview/pull/91

There are also github actions to verify the consistency of a CODEOWNERS file.  

* https://github.com/mszostok/codeowners-validator

Some example uses of this checker in:

* https://github.com/tianocore/edk2-codereview/blob/master/.github/workflows/CheckCodeOwnerFiles.yml
* https://github.com/tianocore/edk2-codereview/blob/master/.github/workflows/CheckCodeOwnerMaintainers.yml

If there are CODEOWNERS that have not joined the EDK II Branch Maintainers team, then that need to be fixed first. Wither then need to join or they need to be removed from CODEOWNERS.

**Comment by bcran** [comment]
> If there are CODEOWNERS that have not joined the EDK II Branch Maintainers team, then that need to be fixed first. They either need to join or they need to be removed from CODEOWNERS.

It looks like Jeremy Soller (@jackpot51) hasn't accepted the invite yet, nor has Bibo Mao (@bibo-mao).

**Pull request merged**
